src/Core/KeyValueStorage/Type/DBKeyValueStorage.php

```diff
@@ -0,0 +1,113 @@
+<?php
+/**
+ * @copyright Copyright (C) 2010-2022, the Friendica project
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace Friendica\Core\KeyValueStorage\Type;
+
+use Friendica\Core\Config\Util\ValueConversion;
+use Friendica\Core\KeyValueStorage\Exceptions\KeyValueStoragePersistenceException;
+use Friendica\Database\Database;
+
+/**
+ * A Key-Value storage provider with DB as persistence layer
+ */
+class DBKeyValueStorage extends AbstractKeyValueStorage
+{
+	const DB_KEY_VALUE_TABLE = 'key-value';
+
+	/** @var Database */
+	protected $database;
+
+	public function __construct(Database $database)
+	{
+		$this->database = $database;
+	}
+
+	/** {@inheritDoc} */
+	public function offsetExists($offset): bool
+	{
+		try {
+			return $this->database->exists(self::DB_KEY_VALUE_TABLE, ['k' => $offset]);
+		} catch (\Exception $exception) {
+			throw new KeyValueStoragePersistenceException(sprintf('Cannot check storage with key %s', $offset), $exception);
+		}
+	}
+
+	/** {@inheritDoc} */
+	public function offsetGet($offset)
+	{
+		try {
+			$result = $this->database->selectFirst(self::DB_KEY_VALUE_TABLE, ['v'], ['k' => $offset]);
+
+			if ($this->database->isResult($result)) {
+				$value = ValueConversion::toConfigValue($result['v']);
+
+				// just return it in case it is set
+				if (isset($value)) {
```

Comment:
Why is this condition necessary?

Reply:
It's a copy&paste from the corresponding Config class. I tried to be as compatible as possible to the current implementation. And yes, I wondered myself too ;)